controllers/account/ec2.go

```diff
@@ -628,8 +625,48 @@ func CreateEC2Instance(reqLogger logr.Logger, account *awsv1alpha1.Account, clie
 		}
 
 		// If fedramp, create subnet and set value for RunInstancesInput
-		if config.IsFedramp() {
-			subnetID, err := createSubnet(reqLogger, client, account, managedTags, customerTags, sampleCIDR, sampleVPCID)
+		if !config.IsFedramp() {
+			// Get a list of all VPCs with appropriate tag
+			result, err := client.DescribeVpcs(&ec2.DescribeVpcsInput{
+				MaxResults: aws.Int64(5),
+				Filters: []*ec2.Filter{
+					{
+						Name: aws.String("tag:" + awsv1alpha1.EC2InstanceNameTagKey),
+						Values: []*string{
+							aws.String(awsv1alpha1.EC2InstanceNameTagValue),
+						},
+					},
+					{
+						Name: aws.String("tag-key"),
+						Values: []*string{
+							aws.String(awsv1alpha1.ClusterAccountNameTagKey),
+						},
+					},
+					{
+						Name: aws.String("tag-key"),
+						Values: []*string{
+							aws.String(awsv1alpha1.ClusterNamespaceTagKey),
+						},
+					},
+					{
+						Name: aws.String("tag-key"),
+						Values: []*string{
+							aws.String(awsv1alpha1.ClusterClaimLinkTagKey),
+						},
+					},
+					{
+						Name: aws.String("tag-key"),
+						Values: []*string{
+							aws.String(awsv1alpha1.ClusterClaimLinkNamespaceTagKey),
+						},
+					},
+				},
+			})
+			if err != nil {
+				reqLogger.Error(err, "Error while describing VPCs")
+				controllerutils.LogAwsError(reqLogger, "Error finding vpcID", nil, err)
+			}
+			subnetID, err := createSubnet(reqLogger, client, account, managedTags, customerTags, sampleCIDR, *result.Vpcs[0].VpcId)
```

Comment:
We should first nil check these, I presume if no VPCs match then we get an empty slice back. For example, `if result != nil && len(*result.Vpcs) > 0 {}`